source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -1015,15 +1015,10 @@ void ShaderGraph::optimize(GenContext& context)
     {
         if (node->hasClassification(ShaderNode::Classification::CONSTANT))
         {
-            // Constant nodes can be removed by assigning their value downstream
-            // But don't remove it if it's connected upstream, i.e. it's value
-            // input is published.
-            ShaderInput* valueInput = node->getInput(0);
-            if (!valueInput->getConnection())
```

Comment:
Previously constant nodes where not removed if connected upstream. I can't find a good reason for this restriction. 

A constant node results in an extra local variable declaration in generated code, which In GLSL triggered an issue if the input is of filename type (a texture) since GLSL don't allow local variables of sampler type.

By always removing (bypassing) constant nodes we avoid the extra local variable, and avoid the GLSL issue.